### ISSUE TITLE
podman run with a flag "--cap-drop all" fails on Ubuntu 18.04 with an error "starting container process caused "error adding seccomp rule for syscall socket: requested action matches default action of filter": OCI runtime error"

### ISSUE THREAD

**Issue by alexshkatov** [containers/podman#8055]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**
Background: I'd like to have a full control over the container capabilities, so my approach is to drop all the capabilities and then to add some of them explicitly.

Problem: When trying to execute `podman run` with a flag `--cap-drop all` on **Ubuntu 18.04** machine, the command fails with :
```
$ sudo podman run --cap-drop all --cap-add DAC_OVERRIDE --cap-add SETGID --cap-add SETUID ubuntu:18.04 uname -a
Error: container_linux.go:349: starting container process caused "error adding seccomp rule for syscall socket: requested action matches default action of filter": OCI runtime error
```
Whithout the flag, it works fine:
```
$ sudo podman run ubuntu:18.04 uname -a
Linux fd4da71769aa 5.4.0-1028-gcp #29~18.04.1-Ubuntu SMP Mon Oct 5 16:41:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

**Steps to reproduce the issue:**

1. Install Ubuntu 18.04 machine
```
$ uname -a
Linux ashkatov-ubuntu-18-04 5.4.0-1028-gcp #29~18.04.1-Ubuntu SMP Mon Oct 5 16:41:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
2. Install the latest podman according to the official instructions at https://podman.io/getting-started/installation.html

3. Run 
```
$ sudo podman run --cap-drop all --cap-add DAC_OVERRIDE --cap-add SETGID --cap-add SETUID ubuntu:18.04 uname -a
```
**Describe the results you received:**
```
Error: container_linux.go:349: starting container process caused "error adding seccomp rule for syscall socket: requested action matches default action of filter": OCI runtime error
```

**Describe the results you expected:**
```
Linux fd4da71769aa 5.4.0-1028-gcp #29~18.04.1-Ubuntu SMP Mon Oct 5 16:41:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:      2.1.1
API Version:  2.0.0
Go Version:   go1.15.2
Built:        Thu Jan  1 00:00:00 1970
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.16.1
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: 'conmon: /usr/libexec/podman/conmon'
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.20, commit: '
  cpus: 2
  distribution:
    distribution: ubuntu
    version: "18.04"
  eventLogger: journald
  hostname: ashkatov-ubuntu-18-04
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1010
      size: 1
    - container_id: 1
      host_id: 755360
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1009
      size: 1
    - container_id: 1
      host_id: 755360
      size: 65536
  kernel: 5.4.0-1028-gcp
  linkmode: dynamic
  memFree: 389570560
  memTotal: 4126576640
  ociRuntime:
    name: runc
    package: 'runc: /usr/sbin/runc'
    path: /usr/sbin/runc
    version: 'runc version spec: 1.0.1-dev'
  os: linux
  remoteSocket:
    path: /run/user/1009/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: 'slirp4netns: /usr/bin/slirp4netns'
    version: |-
      slirp4netns version 1.1.4
      commit: unknown
      libslirp: 4.3.1-git
      SLIRP_CONFIG_VERSION_MAX: 3
  swapFree: 0
  swapTotal: 0
  uptime: 6h 33m 36.58s (Approximately 0.25 days)
registries:
  search:
  - docker.io
  - quay.io
store:
  configFile: /home/ashkatov/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: vfs
  graphOptions: {}
  graphRoot: /home/ashkatov/.local/share/containers/storage
  graphStatus: {}
  imageStore:
    number: 0
  runRoot: /run/user/1009/containers
  volumePath: /home/ashkatov/.local/share/containers/storage/volumes
version:
  APIVersion: 2.0.0
  Built: 0
  BuiltTime: Thu Jan  1 00:00:00 1970
  GitCommit: ""
  GoVersion: go1.15.2
  OsArch: linux/amd64
  Version: 2.1.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
Listing... Done
podman/unknown,now 2.1.1~2 amd64 [installed]
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**
The output above was collected on a virtual machine on GCP, but the issue happens on other platforms also.

**Comment by mheon** [comment]
Can you get your `runc` version? We saw this issue before on Fedora with a too-old `runc`.

**Comment by alexshkatov** [comment]
```
$ runc --version
runc version spec: 1.0.1-dev
```
This is a standard Ubuntu 18.04 image on Google Cloud Platform.

**Comment by mheon** [comment]
If that's the standard Ubuntu `runc` binary it's definitely too old. I think we provide a runc in the Kubic repos that should be new enough (or, alternatively, `crun` does not have the issue).

**Comment by alexshkatov** [comment]
Thank you for the quick response!
Seems like Kubic repos don't provide runc for Ubuntu, only Centos (https://build.opensuse.org/package/show/devel:kubic:libcontainers:stable/runc).
I followed Podman official installation page (https://podman.io/getting-started/installation.html) and compiled runc from the source.
Now Podman works as expected
```
$ runc --version
runc version 1.0.0-rc92+dev
commit: d8bfd6c1ca961cb308d56d6e997de5b8d0c4004d
spec: 1.0.2-dev
go: go1.15.2
libseccomp: 2.4.3
$ sudo podman run --cap-drop all --cap-add DAC_OVERRIDE --cap-add SETGID --cap-add SETUID ubuntu:18.04 uname -a
Linux 42bd668c6890 5.4.0-1028-gcp #29~18.04.1-Ubuntu SMP Mon Oct 5 16:41:43 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
So the issue is resolved.
I strongly recommend to update the official documentation since it's misleading.
According to https://podman.io/getting-started/installation.html, `1.0.1-dev` runc installed on my Ubuntu is sufficient:

> Version 1.0.0-rc4 is the minimal requirement, which is available in Ubuntu 18.04 already. To double-check, runc --version should produce at least spec: 1.0.1

but definitely it is not supported.
